src/components/watermark/watermark.tsx

```diff
@@ -67,6 +68,46 @@ class Watermark extends Component<any, any> {
       this.setState({show: true});
       this.props.eventManager.listenOnce(player, player.Event.PLAYING, onPlaying);
     });
+    this._handleLogoUrl();
+  }
+
+  
+  /**
+   * handles the logo url
+   * if the url contains ${entryId}, then replace it with the played entry id
+   *
+   * @returns {void}
+   * @memberof Logo
+   */
+  private _handleLogoUrl(): void {
```

Comment:
change the function names to handleWatermark... and _setWatermark... i think
it would be clearer that way